datacube_stats/utils/__init__.py

```diff
@@ -96,6 +96,21 @@ def to_float(da):
     return data.apply(to_float, keep_attrs=True)
 
 
+def restore_nodata(var_name, var, measurements):
+    assert isinstance(var, xarray.DataArray)
+
+    values = var.values
+
+    if da_is_float(var):
+        for measurement in measurements:
+            if measurement['name'] == var_name:
+                nodata = measurement['nodata']
```

Comment:
We should not assume that `nodata` field is set.

If it's not set no conversion should be happening.

Storing float32 data with `NaN`s is perfectly valid use-case.

Reply:
Changed so that no conversion happens when no `nodata`.